_test/test_utilities/test_deterministic_pseudorandom_sequence.m

```diff
@@ -0,0 +1,68 @@
+classdef test_deterministic_pseudorandom_sequence < TestCase
+%=========================================================================
+% Test(s) to ensure that the "pseudo-random" sequence generated by
+% the noisify_test_rand class over multiple calls to one of its objects
+% for the myrand method yield the same sequence as one created directly
+% over an explicit single range.
+%
+% Cloned from "test_rand_like.m" without altering that script's 
+% particular idiosyncrasies.
+%=========================================================================
+
+    properties
+        % Current path for noisify_test_rand.
+        % Needs to be added and removed from the path to make the
+        % test function
+        deterministic_pseudorandom_sequence_path = '..\shared';
+    end
+    methods
+        
+        % constructor for the test class - constructs base
+        % TestClass instance and minimally displays the name argument.
+        function this=test_deterministic_pseudorandom_sequence(name)
+            this = this@TestCase(name);
+            disp(name)
+        end
+        
+        % test function to verify operation of noisify_test_rand
+        function test_if_deterministic_pseudorandom_sequence_is_ok(this)
+            % find noisify_test_rand
+            addpath(this.deterministic_pseudorandom_sequence_path);
```

Comment:
If this test fails then the `rmpath` call on line 65 won't be reached and the path added here will stay on the path. The solution to this is to use a [cleanup object](https://www.mathworks.com/help/matlab/ref/oncleanup.html). 

E.g. 
```matlab
addpath(this.deterministic_pseudorandom_sequence_path);
cleanup = onCleanup(@() rmpath(this.deterministic_pseudorandom_sequence_path);
```

Reply:
This is useful, thanks. But in what sense does it remain on the path? When the test concludes, what environment has this path? I was expecting that the environment would be destroyed at end of test.

Reply:
Unless I'm mistaken, it's added to the global path for that Matlab session. It's not just added for the scope of the test - you need to manually remove it.